### PR TITLE
added chat session

### PR DESCRIPTION
**Pull request by adeelehsan**

### CODE REVIEW

File: int_tests/test_chat.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    ChatParameters
+
+
+class TestChat(unittest.TestCase):
+
+    def setUp(self):
+        self.addCleanup(self.cleanup)
+        self.client = Factory().build()
+        self.client.corpora.create(name="test-chat", key="test-chat")
+        time.sleep(30)
```
Comment (waqqas-vectara) [inline int_tests/test_chat.py:17]: Consider removing it

File: int_tests/test_chat.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        test_search_1_document = CoreDocument(
+            id="my-doc-id",
+            document_parts=[
+                CoreDocumentPart(
+                    text="Robot Utility Models are trained on a diverse set of environments and objects, and then can "
+                         "be deployed in novel environments with novel objects without any further data or training.",
```
Comment (waqqas-vectara) [inline int_tests/test_chat.py:23]: Consider using triple quotes

**Review by waqqas-vectara — APPROVED** [review]

File: src/vectara/client.py [hunk 3]
```diff
@@ -1,12 +1,96 @@
-from .base_client import BaseVectara, AsyncBaseVectara
+from . import SearchCorporaParameters, GenerationParameters, ChatParameters, ChatStreamedResponse
```
[the remark this answers is not in the document]
Comment (ofermend) [reply src/vectara/client.py:1]: That is usually caught in "make lint" - do we have lint implemented here, and if so did it not catch that?

File: src/vectara/client.py [hunk 4]
```diff
@@ -1,12 +1,96 @@
...
 from vectara.managers.corpus import CorpusManager
 from vectara.managers.upload import UploadManager
 from vectara.managers.document import DocumentManager
 from vectara.utils import LabHelper
 
-from typing import Union, Optional, Callable
+from typing import Union, Iterator
 import logging
 
+from typing import Optional
```
Comment (ofermend) [inline src/vectara/client.py:11]: nit: consider combining the two typing imports into a single line.

File: src/vectara/client.py [hunk 5]
```diff
@@ -1,12 +1,96 @@
...
+            request_timeout_millis: Optional[int] = None,
+            generation: Optional[GenerationParameters] = OMIT,
+            chat_config: Optional[ChatParameters] = OMIT,
+            request_options: Optional[RequestOptions] = None,
+    ):
+        self.client = client
+        self.chat_id = chat_id
+        self.search = search
+        self.generation = generation
+        self.chat_config = chat_config
```
Comment (ofermend) [inline src/vectara/client.py:32]: Do we need chat_config? It's got only a single parameter (whether to store chat or not from the API), so I think it's redundant. When you create a ChatSession - you always want that to be true - wdyt?

File: int_tests/test_chat.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            response_language="eng",
+            citations=CitationParameters(
+                style="none",
+            ),
+            enable_factual_consistency_score=True,
+        )
+        self.chat_params = ChatParameters(store=True)
+        self.request_options = RequestOptions(timeout_in_seconds=100)
+
+    def test_chat(self):
```
Comment (ofermend) [inline int_tests/test_chat.py:55]: Can we test for some of the edge cases?
1. different search parameters (including none provided)
2. different generation parameters (including none provided)

File: src/vectara/client.py [hunk 7]
```diff
@@ -1,12 +1,96 @@
...
+
+class ChatSession:
+    def __init__(
+            self,
+            client,
+            search: SearchCorporaParameters,
+            chat_id: Optional[str] = None,
+            request_timeout: Optional[int] = None,
+            request_timeout_millis: Optional[int] = None,
+            generation: Optional[GenerationParameters] = OMIT,
```
[the remark this answers is not in the document]
Comment (adeelehsan) [reply src/vectara/client.py:24]: default values would be None or whatever server uses for default. From the SDK we pass it as a None.

File: src/vectara/client.py (hunk 5 above)
Comment (adeelehsan) [reply src/vectara/client.py:32]: I think lets keep it because API does have the option to store it or not. Some users might not want to store the chat.
